snorkel/learning_utils.py

```diff
@@ -214,24 +219,33 @@ def fit(self, candidates, gold_labels, **model_hyperparams):
             print "=" * 60
 
             # Train the model
-            self.learner.train(**model_hyperparams)
+            self.model.train(self.X, self.training_marginals, **model_hyperparams)
 
             # Test the model
-            scores   = self.learner.test(candidates, gold_labels, display=False, return_vals=True)
-            p, r, f1 = scores[:3]
+            tp, fp, tn, fn = self.model.score(X_test, L_test, gold_candidate_set, b, set_unlabeled_as_neg, disp=False)
+            p, r = float(len(tp)) / (len(tp) + len(fp)), float(len(tp)) / (len(tp) + len(fn))
+            f1 = 2.0 * (p * r) / (p + r) if (p + r) > 0 else 0
             run_stats.append(list(param_vals) + [p, r, f1])
             if f1 > f1_opt:
-                w_opt      = self.learner.model.w
+                w_opt      = self.model.w
                 param_opts = param_vals
                 f1_opt     = f1
 
         # Set optimal parameter in the learner model
-        self.learner.model.w = w_opt
+        self.model.w = w_opt
 
         # Return DataFrame of scores
         self.results = DataFrame.from_records(run_stats, columns=self.param_names + ['Prec.', 'Rec.', 'F1'])
         return self.results
-
+    
+    
+class RandomSearch(GridSearch):
```

Comment:
Minor: I was thinking `RandomSearch` would be over a continuous space of param values, rather than filtering a set of given discreet values as here; I think that would be more convenient because then the user only needs to specify a range rather than a set of values?